### PR TITLE
Ewok0

### PR DESCRIPTION
**Pull request by EgorZhuk**
Changes for reporting team iteration 0. PRD-5021 and BISERVER-1225.
@tmorgner

### CODE REVIEW

**Comment by tmorgner** [comment]
Can you please split this into two pull requests? One for the two PRD-5021 cases and one for the rest of the commits. The "merge" commit at the end is strange, it only modifies files that are not even touched by the PRD-5021 commits. It seems that this "merge" is a hidden bug fix and should be marked as such.

**Comment by tmorgner** [comment]
General: In the Pentaho-platform and its plugins, the naming convention is that interfaces are prefixed with an upper-case I (basically the dot-net practice). Thus the interfaces in the async package should be named "IAsyncReportState" and so on. 

Second: PentahoAsyncExecutor should have an interface with its public methods defined there. Create an IPentahoAsyncExecutor interfce. Then in spring, use the simple classname a bean-id (ie "IPentahoAsyncExecutor"). You then allow OEMs to implement their own strategy if needed, and follow good Pentaho coding practice.

File: src/org/pentaho/reporting/platform/plugin/JobManager.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  private boolean isSuportAsync = true;
+
+  public JobManager( boolean isSuportAsync ) {
+    if ( !isSuportAsync ) {
+      logger.info( "JobManager initialization: async mode marked as disabled." );
+    }
+    this.isSuportAsync = isSuportAsync;
+  }
+
+  @GET public Response getEcho() {
```
Comment (tmorgner) [inline src/org/pentaho/reporting/platform/plugin/JobManager.java]: Is this used? If not, then please remove that.

File: src/org/pentaho/reporting/platform/plugin/staging/AsyncJobFileStagingHandler.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+  private static final Log logger = LogFactory.getLog( AsyncJobFileStagingHandler.class );
+
+  private TrackingOutputStream fileTrackingStream;
+  // package private for testing purpose
+  File tmpFile;
+
+  private static String stringParentDir = ".";
+
+  public AsyncJobFileStagingHandler( OutputStream outputStream, IPentahoSession userSession, String stringParentDir ) throws IOException {
```
Comment (tmorgner) [inline src/org/pentaho/reporting/platform/plugin/staging/AsyncJobFileStagingHandler.java]: If you dont use the output stream, then dont pass it in. The only time this class is instantiated is in a manual constructor call - so delete that parameter. 

Initialize() will fail if stringParentDir is null. This variable is only ever set in the constrcutor. So do that check here, not in initialize(), and thus fail early!

File: src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  private ILogger logger;
+
+  private SimpleReportingComponent reportComponent;
+  private AsyncJobFileStagingHandler handler;
+  private String url;
+
+  private String userSession = "";
+  private String userName = "";
+  private String insnaceId = "";
```
Comment (tmorgner) [inline src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java]: Typo: should be "instanceId".

File: src/org/pentaho/reporting/platform/plugin/staging/AsyncJobFileStagingHandler.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+  public InputStream getStagingContent() throws FileNotFoundException {
+    return new StagingInputStream( new FileInputStream( tmpFile ) );
+  }
+
+  @Override
+  public void close() {
+    this.closeQuietly();
+  }
+
+  @Override
```
Comment (tmorgner) [inline src/org/pentaho/reporting/platform/plugin/staging/AsyncJobFileStagingHandler.java]: Actually, you dont need the TrackingStream at all. Just ask your tmp-file for its file length. This way you also wont hold on to the streams for longer than necessary.

**Comment by EgorZhuk** [comment]
Will split into two requests.

**Pull request closed without merging**